### ISSUE TITLE
[AssetPlugin] Image focal point support

### ISSUE THREAD

**Issue by michaelbromley** [vendure-ecommerce/vendure#93]
The AssetPlugin supports automatic cropping as provided by [Sharp](https://sharp.dimens.io/en/stable/api-resize/#resize) and makes a guess at the "important" point of the image according to an algorithm.

Sometimes it does not make optimum choices, so it would be good to allow the user to manually set a focal point and store this as a coordinate with the Asset entity.

Sharp can then use this coordinate data when generating a cropped image: https://github.com/lovell/sharp/issues/1198#issuecomment-384415103

For an idea of the UI see https://gentics.github.io/gentics-ui-image-editor/

**Comment by michaelbromley** [comment]
## Focal point in search results

This is now implemented on a development branch, and works very well. The last missing piece is that the [`SearchResult` type](https://github.com/vendure-ecommerce/vendure/blob/master/packages/core/src/api/schema/type/product-search.type.graphql#L20) references the product image only as a string to the URL, not as an `Asset` relation. The reason for this is that the search operation needs to be as fast as possible, so by design the `search_index_item` table is de-normalized and contains *no relations*.

So how to get the focal point data for a search? This is definitely required, since the `search` query will drive the majority of product listings, and these in turn are usually where cropped thumbnails are used.

### Solution 1: Asset relations

I did a test where I added relations to the related product & productVariant Assets in the `SearchIndexItem` entity. Then I ran the "search-and-checkout" load test on the 1k product dataset. 

```
current implementation:  24.9 req/s
with asset relations:    11.4 req/s
```
So that is a no-go!

### Solution 2: Inline focal point data
The next solution is to simply add 2 new fields to the `SearchResult` type: `productPreviewFocalPoint: Coordinate` and `productVariantPreviewFocalPoint: Coordinate`. These will contain the focalPoints as `simple-json` columns.

The implication of this solution is that we now need to listen for any changes to `Asset` entities, in case the `focalPoint` was updated. If so we need to update the appropriate index rows.

In order to do this efficiently, we should also store the ID of the asset in question in the search index.

**Comment by michaelbromley** [comment]
Implemented in 0.9.0
